controllers/persistent-volume-claim/controller_test.go

```diff
@@ -0,0 +1,163 @@
+package persistent_volume_claim_test
+
+import (
+	"context"
+	"github.com/openshift/lvm-operator/controllers"
+	persistent_volume_claim "github.com/openshift/lvm-operator/controllers/persistent-volume-claim"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"reflect"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestPersistentVolumeClaimReconciler_Reconcile(t *testing.T) {
+	defaultNamespace := "openshift-storage"
+
+	tests := []struct {
+		name                     string
+		req                      controllerruntime.Request
+		client                   *fake.ClientBuilder
+		want                     controllerruntime.Result
+		wantErr                  bool
+		expectNoStorageAvailable bool
+	}{
+		{
+			name: "testing set deletionTimestamp",
+			req: controllerruntime.Request{NamespacedName: types.NamespacedName{
+				Namespace: defaultNamespace,
+				Name:      "test-deletionTimestamp",
+			}},
+			client: fake.NewClientBuilder().WithObjects(&v1.PersistentVolumeClaim{
+				TypeMeta: metav1.TypeMeta{},
+				ObjectMeta: metav1.ObjectMeta{Namespace: defaultNamespace, Name: "test-deletionTimestamp",
+					DeletionTimestamp: &metav1.Time{Time: time.Now()}},
+				Spec:   v1.PersistentVolumeClaimSpec{},
+				Status: v1.PersistentVolumeClaimStatus{},
+			}),
+			want: controllerruntime.Result{
+				Requeue:      false,
+				RequeueAfter: 0,
+			},
+		},
+		{
+			name: "testing empty storageClassName",
+			req: controllerruntime.Request{NamespacedName: types.NamespacedName{
+				Namespace: defaultNamespace,
+				Name:      "test-emptyStorageClassName",
+			}},
+			client: fake.NewClientBuilder().WithObjects(&v1.PersistentVolumeClaim{
+				TypeMeta:   metav1.TypeMeta{},
+				ObjectMeta: metav1.ObjectMeta{Namespace: defaultNamespace, Name: "test-emptyStorageClassName"},
+				Spec:       v1.PersistentVolumeClaimSpec{},
+				Status:     v1.PersistentVolumeClaimStatus{},
+			}),
+			want: controllerruntime.Result{
+				Requeue:      false,
+				RequeueAfter: 0,
+			},
+		},
+		{
+			name: "testing non-applicable storageClassName",
```

Comment:
I wouldn't do non-applicable here, I'd just say `"testing empty storageClassName"`

Reply:
its not empty though, the empty case is already covered (see above). This case is for a storage class name that is not empty, but also not prefixed with `controllers.StorageClassPrefix`.

Reply:
Just to be safe I also changed the object names in the test cases to make the distinction clearer, hope that resolves the uncertainty